web/src/model/topology.ts

```diff
@@ -183,20 +196,10 @@ export const generateNode = (
   t: TFunction,
   k8sModels: { [key: string]: K8sModel }
 ): NodeModel => {
-  const id = getTopologyNodeId(data.type, data.namespace, data.name, data.addr, data.host);
-  const label = data.name
-    ? data.name
-    : data.addr
-    ? data.addr
-    : data.namespace
-    ? data.namespace
-    : data.host
-    ? data.host
-    : data.displayName
-    ? data.displayName
-    : '';
+  const id = getTopologyNodeId(data.resourceKind, data.namespace, data.name, data.addr, data.host);
+  const label = data.displayName || data.name || data.addr || 'DEBUG ME';
```

Comment:
I'll remove the "DEBUG ME" before merging, but I wanted to leave it for a while during testing

Reply:
done (5c8bbfb4413713dfb9c4c86cb6babc7f479dd364)